File: backend/src/main/java/com/hrms/entity/Employee.java
package com.hrms.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "employees")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class Employee extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    @Email
    private String email;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false, unique = true)
    private String employeeId;

    @Column(nullable = false)
    private String designation;

    @Column(nullable = false)
    private String department;

    @Column(nullable = false)
    private String dateOfJoining;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EmployeeStatus status;

    @Column(length = 500)
    private String profilePhotoUrl;

    // Relationships
    @OneToOne(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private PersonalDetails personalDetails;

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Address> addresses = new ArrayList<>();

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Education> education = new ArrayList<>();

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Employment> employment = new ArrayList<>();

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<FamilyMember> family = new ArrayList<>();

    @OneToOne(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private BankDetails bankDetails;

    @OneToOne(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private PFDetails pfDetails;

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Document> documents = new ArrayList<>();

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Asset> assets = new ArrayList<>();
}

enum EmployeeStatus {
    ACTIVE, INACTIVE, ON_LEAVE, TERMINATED
}

File: frontend/src/types/employee.ts
// Employee Domain Types

export type EmployeeStatus = "ACTIVE" | "INACTIVE" | "ON_LEAVE" | "TERMINATED";

export interface Employee {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  employeeId: string;
  designation: string;
  department: string;
  dateOfJoining: string;
  status: EmployeeStatus;
  profilePhotoUrl?: string;
  personalDetails?: PersonalDetails;
  addresses?: Address[];
  education?: Education[];
  employment?: Employment[];
  family?: FamilyMember[];
  bankDetails?: BankDetails;
  pfDetails?: PFDetails;
  documents?: Document[];
  assets?: Asset[];
}

export interface PersonalDetails {
  id: string;
  dateOfBirth: string;
  gender: "MALE" | "FEMALE" | "OTHER";
  bloodGroup?: string;
  maritalStatus: "SINGLE" | "MARRIED" | "DIVORCED" | "WIDOWED";
  nationality: string;
  aadharNumber?: string;
  panNumber?: string;
  passportNumber?: string;
}

export interface Address {
  id: string;
  type: "PERMANENT" | "CURRENT" | "EMERGENCY";
  street: string;
  city: string;
  state: string;
  country: string;
  zipCode: string;
  isPrimary: boolean;
}

export interface Education {
  id: string;
  degree: string;
  fieldOfStudy: string;
  school: string;
  startDate: string;
  endDate: string;
  description?: string;
  certificateUrl?: string;
}

export interface Employment {
  id: string;
  jobTitle: string;
  department: string;
  startDate: string;
  endDate?: string;
  isCurrentJob: boolean;
  reportingManager?: string;
  salary?: number;
  employmentType: "FULL_TIME" | "PART_TIME" | "CONTRACT" | "INTERN";
}

export interface FamilyMember {
  id: string;
  name: string;
  relationship: "SPOUSE" | "CHILD" | "PARENT" | "SIBLING" | "OTHER";
  dateOfBirth: string;
  mobileNumber?: string;
  dependentStatus?: "DEPENDENT" | "NOT_DEPENDENT";
}

export interface BankDetails {
  id: string;
  accountHolderName: string;
  accountNumber: string;
  ifscCode: string;
  bankName: string;
  branchName: string;
  accountType: "SAVINGS" | "CURRENT";
}

export interface PFDetails {
  id: string;
  uan: string;
  pfNumber: string;
  pfJoiningDate: string;
  employeeContribution: number;
  employerContribution: number;
}

export interface Document {
  id: string;
  type: "AADHAAR" | "PAN" | "PASSPORT" | "LICENSE" | "VISA" | "OTHER";
  fileName: string;
  fileUrl: string;
  expiryDate?: string;
  uploadedAt: string;
}

export interface Asset {
  id: string;
  type: "LAPTOP" | "MOBILE" | "TABLET" | "ID_CARD" | "ACCESS_CARD" | "OTHER";
  serialNumber: string;
  allocationDate: string;
  notes?: string;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data?: T;
  error?: string;
}

export interface PaginatedResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}
